docs/source/torch.compiler_dynamic_shapes.md

```diff
@@ -1,129 +1,474 @@
 
-C++ is fairly similar:
+The `mark_unbacked` function marks a tensor dimension as unbacked. It is unlikely
+to be the tool you need, but it could be useful if the specialization occurs inside
+a condition `guard_size_oblivious(x)`, and if using it removes the specialization.
+Ensure it fixes the specialization and does not introduce a data-dependent error
+that converts to a graph break at or before the specialization location
+you are trying to  avoid. It might be better to use the next option.
 
-- c10::SymInt/SymFloat/SymBool: user-visible classes that simulate int/float/bool.
-- c10::SymNode/SymNodeImpl: analogous to SymNode
-- There is no ShapeEnv in C++; for ease of debugging, the entire symbolic reasoning apparatus is in Python.
+#### backed_size_oblivious
 
-When you write code that is traceable with `make_fx`, it must be able to deal with SymInt/SymFloat/SymBool flowing through it. [The dynamic shapes manual](https://docs.google.com/document/d/1GgvOe7C8_NVOMLOCwDaYV1mXXyHMXY7ExoewHqooxrs/edit#heading=h.fh8zzonyw8ng) gives some guidance for how to do this.
+Use `backed_size_oblivious` if you want to avoid specialization
+during `guard_size_oblivious`and aim to minimize compilations and
+specialization. Set a flag to treat backed as unbacked for all checks in the code
+that participate in size-oblivious reasoning, which avoids
+0/1 specialization for backed elements.
 
-## DimDynamic policy
+## Reducing Compilations: Step by Step
 
-Symbolic reasoning:
+If you have a model that you can run on your master job and have a `tlparse`,
+here's whatyou should do next:
 
-- Value ranges
-- Sympy usage notes
-- Constraints
-- DimDynamic/Constraint
+### Step 1: Mark Dynamic Elements
 
-## Unbacked SymInts
+The first step is to reduce initial compilations that are eventually optimized away
+by automatic dynamic or PGO. This is straightforward because we know it will work
+upfront. If, in one run, a frame starts with static graphs and converges to
+dynamic graphs, and if you notice a reduction in the number of compiled
+frames in a second (warm) PGO-enabled run, it's likely due to this optimization.
 
-To resolve control flow, we check the hint, aka actual value, of a symbolic integer to determine which branch to go. However, in some cases, we may not have a hint: so-called unbacked symbolic integers arise when a size variable emerges from a data-dependent operation like `.nonzero()` or `.item()`. It is illegal to perform control flow on these symbolic integers, so we must graph break on these operations.
+This is a two-step process:
 
-Naively implemented, this is too restrictive: most PyTorch programs will immediately fail if you try to do anything with unbacked symbolic integers. Here are the most important enhancements to make this actually work:
+1. Find elements marked as dynamic by PGO or automatic dynamic.
+2. Mark them as dynamic using one of the {ref}`user_annotations` tools.
+   Using {ref}`dynamic_sources_allow_list` is easiest.
```

Comment:
I would omit the second sentence. While it might be easier, admittedly it is a bit clumsy since it's set globally.

Reply:
removed